docs/contributing/semgrep-contributing.md

```diff
@@ -127,6 +127,38 @@ make rebuild
 
 See the Makefile in `cli/`
 
+## Adding python packages to `semgrep`
+
+While developing you may find that you would like to add a package dependency. Semgrep uses mypy to do static type-checking of its python code, so when you add a new python package, you will also need to add typing stubs for that package. You can add the package you want along with its typing package in three steps. For example, suppose you would like to add the package `pyyaml` to Semgrep.
+
+First, you will need to install the corresponding package with typing stubs. For our `pyyaml` example, the corresponding package is `types-pyyaml`.
+
+```
+pipenv install --dev types-pyyaml
```

Comment:
You don't actually need to install types-pyyaml in pipenv. Installing it in the pre-commit config is what matters for mypy

Reply:
@brendongo isn't pipenv install needed for something, though? Like the code won't run locally until I do the pipenv install of either types-pyyaml or just pyyaml, and I thought you asked me to remove the pyyaml one from my PR so now I'm confused. Can you clarify what we should do, or jump on a call if it's easier to explain that way?

Reply:
Resolved offline